torch/_inductor/mkldnn_lowerings.py

```diff
@@ -526,6 +529,19 @@ def qlinear_unary(
             else:
                 x_zp.realize()
 
+            if isinstance(
+                ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                ir.ComputedBuffer,
+            ) and (x_zp.get_numel() == 0 and x.get_dtype() == torch.int8):
+                # If x_zp is dummy, x is int8 quantized per-tensor and its scale is not reshaped,
+                # then the codegened code would segfault if we don't create a tensor for x_zp.
+                # It's safe to do so since the quantization is symmetric
+                x_zp = V.graph.add_tensor_constant(
+                    torch.tensor(0, dtype=torch.int32), name="x_zp"
+                )
+
+            assert x_zp.get_numel() == 1, "x_zp is incompatible with oneDNN qlinear"
```

Comment:
@leslie-fang-intel, moving this check here increased the coverage of GEMM template support (x_zp was a 1D tensor instead of a 0D tensor for some test_cpp_wrapper.py UTs). We should probably add some uint8 UTs in another PR.

Thanks!

Reply:
@jgong5 @leslie-fang-intel, I added this check (for `x_zp`) outside the check for GEMM template feasibility because it's applicable even when max-autotune is not enabled.

Thanks!